packages/graphql/tests/integration/directives/cypher/cypher-filtering.int.test.ts

```diff
@@ -0,0 +1,1092 @@
+        expect(gqlResult?.data).toEqual({
+            [Movie.plural]: [
+                {
+                    title: "The Matrix",
+                    actors: [
+                        {
+                            name: "Keanu Reeves",
+                        },
+                    ],
+                },
+            ],
+        });
+    });
+
+    test("With sorting", async () => {
+        const Movie = testHelper.createUniqueType("Movie");
+        const Actor = testHelper.createUniqueType("Actor");
+
+        const typeDefs = `
+            type ${Movie} {
+                title: String
+                custom_field: String
+                    @cypher(
+                        statement: """
+                        RETURN "hello world!" AS s
+                        """
+                        columnName: "s"
+                    )
+                actors: [${Actor}!]! @relationship(type: "ACTED_IN", direction: IN)
+            }
+
+            type ${Actor} {
+                name: String
+                movies: [${Movie}!]! @relationship(type: "ACTED_IN", direction: OUT)
+            }
+        `;
+
+        await testHelper.initNeo4jGraphQL({
+            typeDefs,
+        });
+
+        await testHelper.executeCypher(
+            `
+            CREATE (m1:${Movie} { title: "The Matrix" })
+            CREATE (m2:${Movie} { title: "The Matrix Reloaded" })
+            CREATE (a1:${Actor} { name: "Keanu Reeves" })
+            CREATE (a2:${Actor} { name: "Jada Pinkett Smith" })
+            CREATE (a1)-[:ACTED_IN]->(m1)
+            CREATE (a1)-[:ACTED_IN]->(m2)
+            CREATE (a2)-[:ACTED_IN]->(m2)
+            `,
+            {}
+        );
+
+        const query = `
+            query {
+                ${Movie.plural}(
+                    where: { custom_field: "hello world!" }
+                    options: { sort: [{ title: DESC }] }
+                ) {
+                    title
+                    actors {
+                        name
+                    }
+                }
+            }
+        `;
+
+        const gqlResult = await testHelper.executeGraphQL(query);
+
+        expect(gqlResult.errors).toBeFalsy();
+        expect(gqlResult?.data).toEqual({
+            [Movie.plural]: [
+                {
+                    title: "The Matrix Reloaded",
+                    actors: expect.toIncludeSameMembers([
+                        {
+                            name: "Keanu Reeves",
+                        },
+                        {
+                            name: "Jada Pinkett Smith",
+                        },
+                    ]),
+                },
+                {
+                    title: "The Matrix",
+                    actors: [
+                        {
+                            name: "Keanu Reeves",
+                        },
+                    ],
+                },
+            ],
+        });
+    });
+
+    test("Connect filter", async () => {
```

Comment:
*Nitpick* Just for organization purposes, would it be worth splitting this test file into separate files, maybe when tackling complex cases?

Is just that it is hard to track down what features are covered and which aren't in a single test